config/initializers/doorkeeper_openid_connect.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+Doorkeeper::OpenidConnect.configure do
+  issuer do |_resource_owner, _application|
+    "https://www.openstreetmap.org/oauth2"
```

Comment:
Do we really want a hardcoded server name here?
`"https://#{Settings.server_url}/oauth2"`

Reply:
Is the value of `Settings.server_url` set to www.openstreetmap.org, and `Settings.server_protocol` to `https` in production? The template in Settings.yml shows different values.

Based on https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata, the value should be same as what webfinger returns, although this is somewhat conflicting in development environment - `http://localhost:3000/`.

Assuming Settings.server_url and server_protocol are set to https://www.openstreetmap.org, we should use
`"#{Settings.server_protocol}://#{Settings.url}"`

Reply:
Yes, they are set by https://github.com/openstreetmap/chef/blob/eaf2ba261061613a6e23f6514afeb4f50817db66/cookbooks/web/resources/rails_port.rb#L349 so look like:

```yaml
server_protocol: https
server_url: www.openstreetmap.org
```

Reply:
Updated to use `server_protocol` and `server_url` instead of hard-coded value